### PR TITLE
conf-opencc1_1: add Ubuntu-family support

### PR DESCRIPTION
**Pull request by jmid**
...and mark Alpine+Oracle OK to fail on the CI

### CODE REVIEW

**Comment by jmid** [comment]
CI fails on
- Arch
- OpenSuse 15.5
- FreeBSD
- macOS homebrew


d6de78a attempts to fix Arch and 26d1f4f takes a first stab at FreeBSD support.
macOS and OpenSuse 15.5 seem to miss the right linker path (FreeBSD likely also will).

**Comment by jmid** [comment]
CI is down to linker failures on OpenSuse, FreeBSD, and macOS.
e8ea4a8 is a pkg-config attempt to solve it.

**Comment by jmid** [comment]
`pkg-config` in turn may require `PKG_CONFIG_PATH` updates - or fail due to missing .pc-files... :thinking:

File: packages/conf-opencc1_1/conf-opencc1_1.1/opam [hunk 1]
```diff
@@ -10,17 +10,25 @@ tags: [
 ]
 
 build: [
-  ["sh" "-exec" "cc $CFLAGS test.c -l:libopencc.so.1.1"]
```
Comment (mseri) [inline packages/conf-opencc1_1/conf-opencc1_1.1/opam:13]: No opencc package provides PKG config configuration on linux.
I prefer to keep the old version here, even if it fails. People will have to set the CFLAGS manually as they had to do in the past

File: packages/conf-opencc1_1/conf-opencc1_1.1/opam [hunk 2]
```diff
@@ -14,13 +14,21 @@ build: [
...
+  ["opencc"] {os-distribution = "arch"}
   ["app-i18n/opencc"] {os-distribution = "gentoo"}
   ["opencc"] {os-distribution = "fedora"}
   ["opencc"] {os-distribution = "centos"}
   ["opencc"] {os = "macos" & os-distribution = "homebrew"}
+  ["chinese/opencc"] {os = "freebsd"}
+]
+x-ci-accept-failures: [ # RDKit does not exist by default on these distributions
+  "alpine-3.18"
+  "alpine-3.19"
```
Comment (shonfeder) [inline packages/conf-opencc1_1/conf-opencc1_1.1/opam:28]: Looks like we are also seeing failures on `alpine-3.20`, should we add it here?

```suggestion
  "alpine-3.19"
  "alpine-3.20"
```

**Review by shonfeder — COMMENTED** [review]
Just checking in here with some of the CLI errors in case it can help things along.

On FreeBSD, we are seeing

```
[ERROR] The compilation of conf-opencc1_1.1 failed at "sh -exec cc $CFLAGS test.c -l:libopencc.so.1.1".

#=== ERROR while compiling conf-opencc1_1.1 ===================================#
# context              2.2.0 | freebsd/x86_64 | ocaml-base-compiler.4.14.2 | pinned
# path                 ~/.opam/4.14.2/.opam-switch/build/conf-opencc1_1.1
# command              /bin/sh -exec cc $CFLAGS test.c -l:libopencc.so.1.1
# exit-code            1
# env-file             ~/.opam/log/conf-opencc1_1-38711-f44fca.env
# output-file          ~/.opam/log/conf-opencc1_1-38711-f44fca.out
### output ###
# + cc test.c -l:libopencc.so.1.1
# ld: error: unable to find library -l:libopencc.so.1.1
# cc: error: linker command failed with exit code 1 (use -v to see invocation)
```

And similarly, on MacOS:

```
<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
-> retrieved conf-opencc1_1.1  (cached)
[ERROR] The compilation of conf-opencc1_1.1 failed at "sh -exec cc $CFLAGS test.c -l:libopencc.so.1.1".

#=== ERROR while compiling conf-opencc1_1.1 ===================================#
# context              2.2.0~rc1 | macos/x86_64 | ocaml-base-compiler.4.14.2 | pinned
# path                 ~/.opam/4.14.2/.opam-switch/build/conf-opencc1_1.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build sh -exec cc $CFLAGS test.c -l:libopencc.so.1.1
# exit-code            1
# env-file             ~/.opam/log/conf-opencc1_1-11671-c05b68.env
# output-file          ~/.opam/log/conf-opencc1_1-11671-c05b68.out
### output ###
# + cc test.c -l:libopencc.so.1.1
# ld: library ':libopencc.so.1.1' not found
# clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Do you know if these are expected?

**Comment by jmid** [comment]
Taking notes to remind my future self:

For FreeBSD `pkg plist` at https://www.freshports.org/chinese/opencc/ show
```
37. lib/libopencc.so.1.1
```
so it is a matter of passing the right -L path...

The above installation path is typically prefixed though.
Recalling I did something similar for conf-rdkit in #24708, 4fdb015 passes /usr/local/lib.

I then realized the macOS linker did not support the `-l:` syntax to specify a specific library version.
This page https://docstore.mik.ua/orelly/unix3/mac/ch05_04.htm helped with syntax for version specification on macOS.

Finally, the install path prefix in homebrew differs between Intel (/usr/local) and ARM64 (/opt/homebrew)

**Comment by jmid** [comment]
Windows support is indeed out of scope.
Otherwise this is green modulo a lint failure and should thus be good to go.

**Comment by shonfeder** [comment]
Yep, CI is all good. The linting error is not relevant to the changes added here AFAIK, and we don't yet have a way of ignoring windows yet.

LGTM.

**Comment by dinosaure** [comment]
Thanks!

**Pull request merged**